### DISCUSSION TITLE
functions.sh: line 980: http: command not found

### DISCUSSION THREAD

**Discussion by AuZ-ASD** [BlazejosP/huawei-sun2000-API-CLI#36] [Q&A] [answered]
Hi,

first of all: THANKS for developing this... as far as I understand this is exactly what I'm searching for. 
but I have some dificulties geting it to run... but maybe this is because I did it wrong..

I downloaded the zip file and copied it to my debian box
I installed the minimum dependencies as I understan (I dont want the interactive thing... just the cli command which show all the data)
I got the API accound from the company who installed the inverter and put this infos in config.conf and after adding execution rights to functions.sh and fusionsolarapp.sh I tried to run fusionsolarapp.sh but got this error:

> 
> root@munin:/opt/huawei-sun2000-API-CLI-master# ./fusionsolarapp.sh
> 
> Please wait connecting!
> Connection to API
> 
> functions.sh: line 980: http: command not found
> 
> Undefined problem with conection to Huawei Server
> 
> Undefined Error
> \nReturned data:
> root@munin:/opt/huawei-sun2000-API-CLI-master#
> 

if someone could point me to the thing I did wrong this would be nice

Regards

**Comment by januweness** [comment, marked as answer]
Hi,

This looks like a small problem in your Ubuntu, see, e.g.:
https://askubuntu.com/questions/747178/terminal-http-command

Best wishes, JanßUwe

From: AuZ-ASD ***@***.***>
Sent: Tuesday, August 22, 2023 2:39 PM
To: BlazejosP/huawei-sun2000-API-CLI ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [BlazejosP/huawei-sun2000-API-CLI] functions.sh: line 980: http: command not found (Discussion #36)


Hi,

first of all: THANKS for developing this... as far as I understand this is exactly what I'm searching for.
but I have some dificulties geting it to run... but maybe this is because I did it wrong..

I downloaded the zip file and copied it to my debian box
I installed the minimum dependencies as I understan (I dont want the interactive thing... just the cli command which show all the data)
I got the API accound from the company who installed the inverter and put this infos in config.conf and after adding execution rights to functions.sh and fusionsolarapp.sh I tried to run fusionsolarapp.sh but got this error:

`
***@***.***:/opt/huawei-sun2000-API-CLI-master# ./fusionsolarapp.sh

Please wait connecting!
Connection to API

functions.sh: line 980: http: command not found

Undefined problem with conection to Huawei Server

Undefined Error
\nReturned data:
***@***.***:/opt/huawei-sun2000-API-CLI-master#
`

if someone could point me to the thing I did wrong this would be nice

Regards

-
Reply to this email directly, view it on GitHub<https://github.com/BlazejosP/huawei-sun2000-API-CLI/discussions/36>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABTM3AFL4P4ADYSRVAGFVJ3XWSR6TANCNFSM6AAAAAA3Z3KFZA>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.******@***.***>>

This message is intended only for the recipient(s) named above. It may contain proprietary information and/or protected content. Any unauthorised disclosure, use, retention or dissemination is prohibited. If you have received this e-mail in error, please notify the sender immediately. ESA applies appropriate organisational measures to protect personal data, in case of data privacy queries, please contact the ESA Data Protection Officer ***@***.***).

**Comment by AuZ-ASD** [comment]
Thanks!! I just missed to install this dependency :( damn!
